src/main/java/uk/gov/hmcts/reform/coh/controller/RelistingController.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.coh.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.coh.controller.onlinehearing.Relisting;
+import uk.gov.hmcts.reform.coh.domain.OnlineHearing;
+import uk.gov.hmcts.reform.coh.domain.RelistingState;
+import uk.gov.hmcts.reform.coh.service.OnlineHearingService;
+
+import java.util.Optional;
+import java.util.UUID;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(
+    value = "/continuous-online-hearings/{onlineHearingId}/relist",
+    produces = MediaType.APPLICATION_JSON_VALUE
+)
+public class RelistingController {
+
+    @Autowired
+    private OnlineHearingService onlineHearingService;
+
+    @GetMapping
+    public ResponseEntity retrieveRelisting(@PathVariable UUID onlineHearingId) {
+        Optional<OnlineHearing> optionalOnlineHearing = onlineHearingService.retrieveOnlineHearing(onlineHearingId);
+        if (!optionalOnlineHearing.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not found");
+        }
+        OnlineHearing onlineHearing = optionalOnlineHearing.get();
+        String reason = onlineHearing.getRelistReason();
+        RelistingState state = onlineHearing.getRelistState();
+        Relisting relisting = new Relisting(reason, state);
+        return ResponseEntity.ok(relisting);
+    }
+
+    @PostMapping
+    public ResponseEntity createDraft(
+        @PathVariable UUID onlineHearingId,
+        @RequestBody @Valid Relisting body
+    ) {
+        Optional<OnlineHearing> optionalOnlineHearing = onlineHearingService.retrieveOnlineHearing(onlineHearingId);
+        if (!optionalOnlineHearing.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not found");
+        }
+
+        OnlineHearing onlineHearing = optionalOnlineHearing.get();
+
+        if (onlineHearing.getRelistState() == RelistingState.ISSUED) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).body("Already issued");
+        }
+
```

Comment:
1. If the relist is issued to the appellant, then the online hearing also has to be updated
2. I thought we agree to queue the request to follow the the same pattern as issuing of questions/decisions etc

Reply:
1. Online hearing's state is updated right after Relisting's state is changed to `ISSUED`.
2. This feature is coming in the next commits.
3. Will do that after getting feature from 2. point up and running.
4. Reason was supposed to be optional for re-listing or have I missed the spot?
5. Validation is provided by Jackson with `@Valid` annotation.
6. Thanks, I'll add this.
7. Thanks, I'll add this.
8. That reminds me about adding created and updated fields to the response.

Reply:
- `ISSUE_PENDING` state has been implemented so that re-listing will be issued only after successful delivery of the notification.
- Setting directly online hearing as re-listed when updating is now removed.
- Both fields `created` and `updated` are added.